### ISSUE TITLE
`HasManyRepeater` throw `QueryException` when add item

### ISSUE THREAD

**Issue by nuradiyana** [laravel-filament/filament#1209]
### Package

filament/filament

### Package Version

^2.5

### Laravel Version

^8.65

### PHP Version

^8.0

### Bug description

When add new item in `HasManyRepeater` filament throw exception

```php
Illuminate\Database\QueryException
SQLSTATE[22P02]: Invalid text representation: 7 ERROR: invalid input syntax for type bigint: "c92e6ab7-58a3-431e-9498-affe1930fd8f" (SQL: select * from "shop_order_items" where "shop_order_items"."shop_order_id" is null and "shop_order_items"."shop_order_id" is not null and "shop_order_items"."id" = c92e6ab7-58a3-431e-9498-affe1930fd8f limit 1) 
```

This issue came up when I was using postgres as database.

### Steps to reproduce

1. Clone demo from https://github.com/laravel-filament/demo
2. Install composer `composer install`
3. Setting database
4. Before migrate change some table type in `shop_discounts` (error type when seed)
```php
$table->string('type');
$table->unsignedInteger('usage_limit_per_customer')->default(0);
```
5. Running database seed
6. Login and try to add order items

**Comment by danharrin** [comment]
I'm gonna need more info on what you're actually trying to achieve, why you expect this to work, where the UUID is coming from / trying to be saved to etc.

**Comment by danharrin** [comment]
Ah, so the issue is that Postgres doesn't like you checking a `BIGINT` column with a UUID value?

**Comment by nuradiyana** [comment]
Yes, that true, any idea how to handle that? i have an idea, steal from laravel, we check if `keyType` in model string or int, then cast to proper type, but i don't know if this will work.

https://github.com/laravel/framework/blob/8091f07558ff4a890435ff9d25fa9aca0189ad63/src/Illuminate/Database/Eloquent/Builder.php#L233-L235

**Comment by danharrin** [comment]
UUIDs are already strings so that won't make a difference.

The only way I can think of doing this is having to add `new-` to all new UUIDs and then preventing the check from happening if its a new item.

**Comment by danharrin** [comment]
Should be fixed by #1248.
